spacy/tokens/span.pyx

```diff
@@ -181,9 +181,17 @@ cdef class Span:
             return Span(self.doc, start + self.start, end + self.start)
         else:
             if i < 0:
-                return self.doc[self.end + i]
+                token = self.doc[self.end + i]
```

Comment:
```suggestion

```

Reply:
It looks a bit odd to get the token first and check second, a reasonable reader would wonder whether this was a bug as they wouldn't be assured the indexing will be legal.

Reply:
Hmm, I'm not sure I understand what you mean with this suggested change. I was trying to return the same `Doc`-based `IndexError` as before (`E040`) for indices outside the doc and just address the `Span` issue, but if this is what I think you might have meant (?), I agree that it would be nicer to return the same `IndexError` for all cases here.